ObserverDesignPattern/src/ObserverMain.java

```diff
@@ -1,7 +1,14 @@
-//TIP To <b>Run</b> code, press <shortcut actionId="Run"/> or
-// click the <icon src="AllIcons.Actions.Execute"/> icon in the gutter.
 public class ObserverMain {
     public static void main(String[] args) {
+        StocksObservable iphoneStocksObservable = new IphoneStocksObservable();
+        NotificationObserver observer1 = new EmailNotificationObserver(iphoneStocksObservable, "farhanahmad091@gmail.com");
+        NotificationObserver observer2 = new MobileNotificationObserver(iphoneStocksObservable, "8539062684");
 
+        iphoneStocksObservable.add(observer1);
+        iphoneStocksObservable.add(observer2);
```

Comment:
we can increase cohesion by combining observer creation and subject registration at one place.   
![Uploading observer_uml.png…]()